dynamiqs/core/diffrax_solver.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import warnings
+from collections import namedtuple
+
+import diffrax as dx
+
+# from equinox import AbstractVar
+from jaxtyping import PyTree
+
+from .._utils import bexpect
+from ..gradient import Adjoint, Autograd
+from ..result import Result
+from .abstract_solver import BaseSolver
+
+SolverArgs = namedtuple('SolverArgs', ['save_states', 'E'])
+
+
+def save_fn(_t, y, args: SolverArgs):
+    res = {}
+    if args.save_states:
+        res['ysave'] = y
+    if args.E is not None and len(args.E) > 0:
+        res['Esave'] = bexpect(args.E, y)
+    return res
+
+
+class DiffraxSolver(BaseSolver):
+    diffrax_solver: dx.AbstractSolver
+    stepsize_controller: dx.AbstractAdaptiveStepSizeController
+    dt0: float | None
+    max_steps: int
+    term: dx.ODETerm
+
+    def __init__(self, *args):
+        super().__init__(*args)
```

Comment:
```suggestion
```

Reply:
Fun fact, we need this 😅 
```python
import equinox as eqx

class A(eqx.Module):
    a: int

class B(A):
    b: int

#    def __init__(self, *args):
#        super().__init__(*args)

class C(B):
    def __init__(self, *args):
        super().__init__(*args)
        self.b = 2 * self.a

C(1)  # TypeError: __init__() missing 1 required positional argument: 'b'
```
If you uncomment the init of `B` it works as intended...

Maybe I'm missing smthg obvious here, cc @abocquet.

Reply:
Ok that's weird ! Not super important anyway, let's keep it ^^